__tests__/app.test.js

```diff
@@ -144,5 +144,49 @@ describe("All GET /api Tests", () => {
       });
   });
 
+  //POST TEST
+  //task 7
+  describe("POST /api/articles/:article_id/comments", () => {
+    test("status: 201 - should respond with the posted comment", () => {
+        const testComment = {
+            username: "butter_bridge",
+            body: "a new comment",
+        }
+        return request(app)
+            .post("/api/articles/11/comments")
+            .send(testComment)
+            .expect(201)
+            .then(({ body }) => {
+                const { comment } = body
 
+                expect(typeof comment).toBe("object")
+                expect(Object.keys(comment)).toHaveLength(6)
+                expect(comment).toMatchObject({
+                    comment_id: 19,
+                    body: expect.any(String),
+                    article_id: expect.any(Number),
+                    author: expect.any(String),
+                    votes: expect.any(Number),
+                    created_at: expect.any(String),
+                })
+            })
+      })
+
+      test("status: 400 - should respond with an error if username is missing", () => {
+        const testComment = {
+            body: "a comment without a username",
+        }
+        return request(app)
+            .post("/api/articles/11/comments")
+            .send(testComment)
+            .expect(400)
+            .then(({ body }) => {
+                expect(body.message).toBe("Username is required")
+            })
+    })
```

Comment:
This is a good start to error handling for this endpoint, but there are a few other that should be included here as well:

What should happen when all of the properties exist, but the username given does not exist?
E.g `{
            username: "IDontExist"
            body: "a comment with a username that doesnt exist",
        }`
        
What should happen when requesting to post a comment on an article that doesnt exist?
E.g `/api/articles/9999/comments`

What should happen when requesting to post a comment on an invalid article_id?
E.g `/api/articles/banana/comments`